main.go

```diff
@@ -8,7 +8,18 @@ import (
 	"github.com/coredns/coredns/coremain"
 )
 
-var directives = append([]string{"mdns"}, dnsserver.Directives...)
+// We need to inject our plugin after the cache plugin for caching to work
+func findCache() int {
```

Comment:
I find it a bit counter intuitive that findCache gives you the index of the element after "cache"